spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2300,6 +2300,12 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
             expr.children(1),
             inputs,
             (builder, binaryExpr) => builder.setArrayAppend(binaryExpr))
+        case ArraysOverlap(leftArrayExpr, rightArrayExpr) =>
```

Comment:
Would you add a flag to enable/disable?

Reply:
We could use the existing `COMET_CAST_ALLOW_INCOMPATIBLE` config to enable this function until we have comprehensive testing in place.